File: go/83.remove-duplicates-from-sorted-list.go
/*
 * @lc app=leetcode id=83 lang=golang
 *
 * [83] Remove Duplicates from Sorted List
 */
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func deleteDuplicates(head *ListNode) *ListNode {
	if head == nil {
		return nil
	}
	if head.Next == nil {
		return head
	}

	prev := head
	current := head.Next
	for current != nil {
		for current != nil && current.Val == prev.Val {
			current = current.Next
		}
		prev.Next = current
		prev = current
		if current != nil {
			current = current.Next
		}
	}

	return head
}

File: csharp/83.remove-duplicates-from-sorted-list.cs
/*
 * @lc app=leetcode id=83 lang=csharp
 *
 * [83] Remove Duplicates from Sorted List
 */

// @lc code=start
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution 
{
    public ListNode DeleteDuplicates(ListNode head) 
    {
        if (head == null)
        {
            return null;
        }

        ListNode prev, cur, next;
        prev = head;
        cur = head.next;

        while (cur != null)
        {
            if (cur.val == prev.val)
            {
                prev.next = cur.next;
                cur = cur.next;
            }
            else 
            {
                prev = cur;
                cur = cur.next;
            }
        }

        return head;
    }
}
// @lc code=end

